tools/flat-storage/src/commands.rs

```diff
@@ -181,6 +193,109 @@ impl FlatStorageCommand {
+            .expect("Error reading from DB")
+            .expect(format!("Block header not found with {block_hash_vec:?}").as_str())
+    }
+
+    fn set_block_misc(height: BlockHeight, source_store: &Store, target_store: &Store) {
+        let mut db_update = target_store.store_update();
+
+        let header = Self::get_header(height, source_store);
+        let tip = Tip::from_header(&header);
+
+        let col = DBCol::BlockMisc;
+        db_update.set_ser(col, near_store::HEAD_KEY, &tip).expect("Unable to write HEAD_KEY");
+        db_update
+            .set_ser(col, near_store::HEADER_HEAD_KEY, &tip)
+            .expect("Unable to write HEADER_HEAD_KEY");
+        db_update
+            .set_ser(col, near_store::FINAL_HEAD_KEY, &tip)
+            .expect("Unable to write FINAL_HEAD_KEY");
+        db_update.commit().expect("Unable to commit to TestDB");
+    }
+
+    fn get_amend_db(height: BlockHeight, store: &Store) -> Arc<dyn Database> {
+        let db = TestDB::new();
+        let db_store = NodeStorage::new(db.clone()).get_hot_store();
+        Self::set_block_misc(height, store, &db_store);
+        db
+    }
+
+    fn get_init_db(
+        opener: &StoreOpener,
+        home_dir: &PathBuf,
+        near_config: &NearConfig,
+        height: BlockHeight,
+        target_db_path: &Path,
+    ) -> (NodeStorage, Arc<EpochManagerHandle>, Arc<NightshadeRuntime>, ChainStore, Store) {
+        tracing::info!(target: "flat", ?height, ?target_db_path, "get_init_db");
+        let node_storage = {
+            let (base_db, amend_db) = {
+                let node_storage = opener.open().expect("Unable to create NodeStorage");
+                let amend_db = Self::get_amend_db(height, &node_storage.get_hot_store());
+                let base_db = match node_storage.cold_db() {
+                    Some(cold_storage) => {
+                        let cold_storage = cold_storage.clone();
+                        SplitDB::new(node_storage.into_inner(Temperature::Hot), cold_storage)
+                    }
+                    None => node_storage.into_inner(Temperature::Hot),
+                };
+                (base_db, amend_db)
+            };
+            let write_db = {
+                let path = if target_db_path.is_absolute() {
+                    PathBuf::from(target_db_path)
+                } else {
+                    home_dir.join(&target_db_path)
+                };
+                Arc::new(
+                    near_store::db::RocksDB::open(
+                        &path,
+                        &StoreConfig::default(),
+                        Mode::ReadWrite,
+                        near_store::Temperature::Hot,
+                    )
+                    .expect("Unable to open recovery db"),
+                )
+            };
+
+            // This DB can only write to write_db
+            // When reading, it first reads from wirte_db, then amended_db, then base_db
+            let mixed_db = MixedDB::new(
```

Comment:
confirming
- we are creating a mixed db where we are writing to the custom recovery database we are trying to create with the flat storage?
- we are reading from the split storage db but with one tiny change where we are manually setting the final height to our custom value?

Reply:
True and True